File: clients/scala/src/main/scala/komus24/model/MinMaxRangeLong.scala
package komus24.model

import komus24.util.StreamUtil

/**
 * TODO - Document
 *
 * @param min TODO - Document
 * @param max TODO - Document
 */
case class MinMaxRangeLong(min: Long, max: Long) {
    /**
     * Write MinMaxRangeLong to output stream
     */
    def writeTo(stream: java.io.OutputStream): scala.Unit = {
        StreamUtil.writeLong(stream, min)
        StreamUtil.writeLong(stream, max)
    }

    /**
     * Get string representation of MinMaxRangeLong
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("MinMaxRangeLong { ")
        stringBuilder.append("min: ")
        stringBuilder.append(min)
        stringBuilder.append(", ")
        stringBuilder.append("max: ")
        stringBuilder.append(max)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object MinMaxRangeLong {
    /**
     * Read MinMaxRangeLong from input stream
     */
    def readFrom(stream: java.io.InputStream): MinMaxRangeLong = MinMaxRangeLong(
        StreamUtil.readLong(stream),
        StreamUtil.readLong(stream)
    )
}

File: clients/java/src/main/java/komus24/model/MinMaxRangeLong.java
package komus24.model;

import komus24.util.StreamUtil;

/**
 * Range of values
 */
public class MinMaxRangeLong {
    /**
     * Minimal value
     */
    private long min;

    /**
     * Minimal value
     */
    public long getMin() {
        return min;
    }

    /**
     * Minimal value
     */
    public void setMin(long value) {
        this.min = value;
    }
    /**
     * Maximal  value
     */
    private long max;

    /**
     * Maximal  value
     */
    public long getMax() {
        return max;
    }

    /**
     * Maximal  value
     */
    public void setMax(long value) {
        this.max = value;
    }

    public MinMaxRangeLong(long min, long max) {
        this.min = min;
        this.max = max;
    }

    /**
     * Read MinMaxRangeLong from input stream
     */
    public static MinMaxRangeLong readFrom(java.io.InputStream stream) throws java.io.IOException {
        long min;
        min = StreamUtil.readLong(stream);
        long max;
        max = StreamUtil.readLong(stream);
        return new MinMaxRangeLong(min, max);
    }

    /**
     * Write MinMaxRangeLong to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeLong(stream, min);
        StreamUtil.writeLong(stream, max);
    }

    /**
     * Get string representation of MinMaxRangeLong
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("MinMaxRangeLong { ");
        stringBuilder.append("min: ");
        stringBuilder.append(String.valueOf(min));
        stringBuilder.append(", ");
        stringBuilder.append("max: ");
        stringBuilder.append(String.valueOf(max));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}